src/language-html/print/block.js

```diff
@@ -0,0 +1,123 @@
+import { group, hardline, indent } from "../../document/builders.js";
+import { printChildren } from "./children.js";
+
+let uid = 0;
+
+const settings = {
+  if: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: ["else if", "else"],
+  },
+  "else if": {
+    withExpression: true,
+    isFollowingBlock: true,
+    followingBlocks: ["else if", "else"],
+  },
+  else: {
+    withExpression: false,
+    isFollowingBlock: true,
+    followingBlocks: [],
+  },
+
+  switch: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: [],
+  },
+  case: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: [],
+  },
+  default: {
+    withExpression: false,
+    isFollowingBlock: false,
+    followingBlocks: [],
+  },
+
+  for: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: ["empty"],
+  },
+  empty: {
+    withExpression: false,
+    isFollowingBlock: true,
+    followingBlocks: [],
+  },
+
+  defer: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: ["placeholder", "error", "loading"],
+  },
+  placeholder: {
+    withExpression: false,
+    isFollowingBlock: true,
+    followingBlocks: ["placeholder", "error", "loading"],
+  },
+  error: {
+    withExpression: false,
+    isFollowingBlock: true,
+    followingBlocks: ["placeholder", "error", "loading"],
+  },
+  loading: {
+    withExpression: true,
+    isFollowingBlock: true,
+    followingBlocks: ["placeholder", "error", "loading"],
+  },
+};
+
+function printBlock(path, options, print) {
+  const { node } = path;
+
+  const setting = settings[node.name];
+  const contents = [];
+
+  if (!setting) {
+    throw new Error("Unknown block name: " + node.name);
+  }
+
+  if (
+    setting.isFollowingBlock &&
+    path.previous &&
+    path.previous.type === "block"
+  ) {
+    contents.push("} ");
+  }
+
+  contents.push(`@${node.name}`);
+
+  if (setting.withExpression) {
+    const expression = fetchExpression(node);
+    contents.push(` (${expression})`);
+  }
+
+  contents.push(" {");
+
+  const children = printChildren(path, options, print);
+  contents.push(indent([hardline, children]));
+
+  if (shouldCloseBlock(node, setting.followingBlocks)) {
+    contents.push(hardline, "}");
+  }
+
+  return group(contents, {
+    id: Symbol("block-" + ++uid),
+    shouldBreak: true,
+  });
+}
+
+export { printBlock };
+
+function shouldCloseBlock(node, names) {
+  return names.length === 0 || !node.next || !names.includes(node.next.name);
+}
+
+function fetchExpression(node) {
+  return node.parameters
+    .filter((parameter) => parameter.type === "blockParameter")
+    .map((parameter) => parameter.expression)
+    .join("; ");
```

Comment:
Do we want allow them to break into multiple lines?

(following use a small printWidth)

**Prettier pr-15606**
[Playground link](https://deploy-preview-15606--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEABAJnAZnATgAgApp8BLWPAQzBlOgG58AHXbOGMAC3xM4gDc8ASnzAAOlAA8lJkwC0YAMwB6AHwSAvhNRMANtTh9dmAuKkBnJpSiqACvrCGIxvET6CCpGABp8MTgj4YLqkYADWPFBkMACEQpLKltbqUBog3iAQTLTQ5siglLi4EADutoUIeSiUuiWUAJ55GQBGuNRh7ADKlAC2cAAy5HDIWDXmcC1t4V1WYOQA5sgwuACuEyDjPaRLq+vmC7pwAIorEPAjY+sAVuYAHp0Hx6fnSKO64xkAjs9wtsVMVRAlHMcigcDgmHQ6RAy0opBCUHmAGEID0epRkEDdLpoftEYcAIIwZakZoreC2PCDMEXd7rTgwHq6ADqnC8cCSjk6lS8pH4XnqmLA5iaIH4awAklBMLBOmBcKRsgTpZ0YPVDrSPiAWBBxsy2kxMSwOXhBNDyONcDA-pR5ujNesrLhLZjrPMVvpcNCWOQYMzSOh-MgAEwABgyrG+pFYNrtGNelwyMEozX9gc4IYyK3GABUU1U3lq4D1mhDIf03StbXAAGIQXDo4kLV3kiAgDQaIA)
<!-- prettier-ignore -->
```sh
--parser angular
--print-width 20
```

**Input:**
<!-- prettier-ignore -->
```html
@defer (on interaction; prefetch on hover) {
<app-c3/>
}
@placeholder {
<span>Placeholder (hover it, then click on it!)</span>
}
```

**Output:**
<!-- prettier-ignore -->
```html
@defer (on interaction; prefetch on hover) {
  <app-c3 />
} @placeholder {
  <span
    >Placeholder
    (hover it, then
    click on
    it!)</span
  >
}

```